### PR TITLE
feat: add missing Darwin implementation for persistence module

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Add missing Darwin implementation for persistence module

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by vitorhugods — APPROVED** [review]
Noice 👌

Just a couple of comments, no blockers from my side.

File: persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/GlobalDatabaseProvider.kt [hunk 1]
```diff
@@ -13,14 +15,36 @@ import com.wire.kalium.persistence.daokaliumdb.LogoutReasonAdapter
...
+                version = schema.version,
+                create = { connection ->
+                    wrapConnection(connection) { schema.create(it) }
+                },
+                upgrade = { connection, oldVersion, newVersion ->
+                    wrapConnection(connection) { schema.migrate(it, oldVersion, newVersion) }
+                },
+                extendedConfig = DatabaseConfiguration.Extended(
+                    basePath = storePath,
+                    foreignKeyConstraints = true
```
[the remark this answers is not in the document]
Comment (typfel) [reply persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/GlobalDatabaseProvider.kt:43]: I wasn't aware it was done in common code so I can remove this.

File: persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/UserDatabase.kt [hunk 2]
```diff
@@ -47,19 +69,24 @@ fun inMemoryDatabase(
             },
             upgrade = { connection, oldVersion, newVersion ->
                 wrapConnection(connection) { schema.migrate(it, oldVersion, newVersion) }
-            }
+            },
+            extendedConfig = DatabaseConfiguration.Extended(
+                foreignKeyConstraints = true
```
[the remark this answers is not in the document]
Comment (typfel) [reply persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/UserDatabase.kt:74]: Yes, I'll remove it.

File: persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/GlobalDatabaseProvider.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (typfel) [reply persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/GlobalDatabaseProvider.kt:43]: Hmm there's is no `GlobalDatabaseBuilder`.

[the remark this answers is not in the document]
Comment (vitorhugods) [reply persistence/src/darwinMain/kotlin/com/wire/kalium/persistence/db/GlobalDatabaseProvider.kt:43]: I stand corrected.

Only the user database has a common implementation. I must've forgotten to revisit this after refactoring the user part.

Then it's absolutely fine the way it is.

**Pull request merged**